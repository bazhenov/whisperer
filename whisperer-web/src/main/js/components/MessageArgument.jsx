import React, { Component } from 'react'
import { SQUASHED_MESSAGE_ARGUMENT_LENGTH } from '../constants'

export default class MessageArgument extends Component {

	constructor(props) {
		super(props);
		this.state = { isOpened: false };
		this.handleClick = this.handleClick.bind(this);
	}

	handleClick(e) {
		this.setState({ isOpened: true });
		e.preventDefault();
	}

	render() {
		const text = this.props.text;
		if (this.state.isOpened) return <span className="argument-value">{text}</span>;
		return <a href="#" onClick={this.handleClick}>{text.substring(0, SQUASHED_MESSAGE_ARGUMENT_LENGTH)}...</a>
	}
}
