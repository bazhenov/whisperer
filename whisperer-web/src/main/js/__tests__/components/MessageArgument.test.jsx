import React from "react"
import MessageArgument from "../../components/MessageArgument.jsx"
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it('MessageArgument', () => {
	const wrapper = mount(
		<MessageArgument text="some text some text some text some text"/>
	);
	expect(toJson(wrapper)).toMatchSnapshot();

	wrapper.find('a').first().simulate('click');
	expect(toJson(wrapper)).toMatchSnapshot();
});
