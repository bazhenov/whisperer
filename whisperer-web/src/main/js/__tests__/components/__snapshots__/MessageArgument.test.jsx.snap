// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageArgument 1`] = `
<MessageArgument
  text="some text some text some text some text"
>
  <a
    href="#"
    onClick={[Function]}
  >
    {
    ...
    }
  </a>
</MessageArgument>
`;

exports[`MessageArgument 2`] = `
<MessageArgument
  text="some text some text some text some text"
>
  <span>
    some text some text some text some text
  </span>
</MessageArgument>
`;
